xrpl/asyncio/transaction/reliable_submission.py

```diff
@@ -22,7 +23,7 @@ class XRPLReliableSubmissionException(XRPLException):
 
 
 async def _wait_for_final_transaction_outcome(
-    transaction_hash: str, client: Client, prelim_result: str
+    transaction_hash: str, client: Client, prelim_result: str, attempts: int
```

Comment:
I would recommend defaulting `attempts` to 0 and then you can increment from there (Otherwise you have an undocumented requirement that users who call the function set it to 0)